cryptography/src/ed25519.rs

```diff
@@ -181,6 +222,39 @@ mod tests {
         )
     }
 
+    fn vector_2() -> (Vec<u8>, Vec<u8>, Vec<u8>, Vec<u8>) {
```

Comment:
It uses the same function signature than `vector_1()` but it could return specific types.
It could be done in another PR, in addition of refactoring `test_sign_and_verify()` args types to be specifics (`PrivateKey`, `PublicKey`, `Signature`). Just as we did in Bls12381 vectors tests (https://github.com/commonwarexyz/monorepo/pull/289#discussion_r1897522306)

Reply:
I think we can do it here given we are making changes all over this file already.